core/block_dragger.js

```diff
@@ -460,7 +460,8 @@ const BlockDragger = class {
 const initIconData = function(block) {
   // Build a list of icons that need to be moved and where they started.
   const dragIconData = [];
-  const descendants = block.getDescendants(false);
+  const descendants =
+      /** @type {!Array<!BlockSvg>} */ (block.getDescendants(false));
```

Comment:
Yeah I can go through and add a bunch of overridden methods on BlockSvg (it already has some) I just wasn't sure if that would be too much.

However, thinking about it again, external devs are going to want to call these functions and have them typed properly. So might as well get it done sooner rather than later.

It does kind of suck that BlockSvg needs so many casts though :/ not a super maintainable pattern.

Reply:
Ok I added some new overrides. Can you give them a quick look before I merge @alschmiedt ?